src/functions-and-arrays.js

```diff
@@ -26,84 +55,122 @@ function sum() {}
 // Level 1: Array of numbers
 const numbersAvg = [2, 6, 9, 10, 7, 4, 1, 9];
 
-function averageNumbers() {}
+function averageNumbers(numbersAvg) {
+    if (!numbersAvg.length) return null;
+    let num = 0;
+    for (let number of numbersAvg) {
+        num += number;
+    }
+    return num / numbersAvg.length;
+};
 
+console.log(averageNumbers(numbersAvg));
 
 // Level 2: Array of strings
+
 const wordsArr = ['seat', 'correspond', 'linen', 'motif', 'hole', 'smell', 'smart', 'chaos', 'fuel', 'palace'];
 
-function averageWordLength() { }
+averageWordLength = wordsArr.join('').length / wordsArr.length
+console.log(averageWordLength);
 
 // Bonus - Iteration #4.1
 function avg() {}
 
 // Iteration #5: Unique arrays
+
 const wordsUnique = [
-  'crab',
-  'poison',
-  'contagious',
-  'simple',
-  'bring',
-  'sharp',
-  'playground',
-  'poison',
-  'communion',
-  'simple',
-  'bring'
+    'crab',
+    'poison',
+    'contagious',
+    'simple',
+    'bring',
+    'sharp',
+    'playground',
+    'poison',
+    'communion',
+    'simple',
+    'bring'
 ];
 
-function uniquifyArray() {}
+function uniquifyArray(wordsUnique) {
+    const uniqueArray = [];
 
+    // Loop through array values
+    for (i = 0; i < wordsUnique.length; i++) {
+        if (uniqueArray.indexOf(wordsUnique[i]) === -1) {
+            uniqueArray.push(wordsUnique[i]);
+        }
+    }
+    return uniqueArray;
+}
 
+const uniqueNames = uniquifyArray(wordsUnique);
+console.log(uniqueNames);
 
 // Iteration #6: Find elements
 const wordsFind = ['machine', 'subset', 'trouble', 'starting', 'matter', 'eating', 'truth', 'disobedience'];
 
-function doesWordExist() {}
+function doesWordExist(wordsFind, word) {
+    if (wordsFind.length === 0) return null;
+    for (let el of wordsFind) {
+        if (el === 'disobedience') return true;
```

Comment:
But one more thing: you never know which words you really have in the array. So you can't compare with specific words. It should work for any array of words.